packages/expo-cli/lib/test/install-plugin.test.js

```diff
@@ -0,0 +1,224 @@
+    await withFixture('blank-00', async (projectRoot) => {
+      const spawn = (cmd, args, opts) => {
+        expect(cmd).toBe('expo')
+        expect(args).toEqual([
+          'install',
+          '@bugsnag/plugin-expo-eas-sourcemaps',
+          '--npm'
+        ])
+        expect(opts).toEqual({ cwd: projectRoot })
+
+        const proc = new EventEmitter()
+        // @ts-ignore
+        proc.stdout = new Readable({
+          read () {
+            this.push('some data on stdout')
+            this.push(null)
+          }
+        })
+        // @ts-ignore
+        proc.stderr = new Readable({
+          read () {
+            this.push('some data on stderr')
+            this.push(null)
+          }
+        })
+        setTimeout(() => proc.emit('close', 0), 10)
+        return proc
+      }
+
+      jest.doMock('child_process', () => ({ spawn }))
+      const installPlugin = require('../install-plugin')
+
+      const msg = await installPlugin(projectRoot, { npm: true })
+      expect(msg).toBe(undefined)
+    })
+  })
+
+  it('should allow forcing install with Yarn', async () => {
+    await withFixture('blank-00', async (projectRoot) => {
+      const spawn = (cmd, args, opts) => {
+        expect(cmd).toBe('expo')
+        expect(args).toEqual([
+          'install',
+          '@bugsnag/plugin-expo-eas-sourcemaps',
+          '--yarn'
+        ])
+        expect(opts).toEqual({ cwd: projectRoot })
+
+        const proc = new EventEmitter()
+        // @ts-ignore
+        proc.stdout = new Readable({
+          read () {
+            this.push('some data on stdout')
+            this.push(null)
+          }
+        })
+        // @ts-ignore
+        proc.stderr = new Readable({
+          read () {
+            this.push('some data on stderr')
+            this.push(null)
+          }
+        })
+        setTimeout(() => proc.emit('close', 0), 10)
+        return proc
+      }
+
+      jest.doMock('child_process', () => ({ spawn }))
+      const installPlugin = require('../install-plugin')
+
+      const msg = await installPlugin(projectRoot, { yarn: true })
+      expect(msg).toBe(undefined)
+    })
+  })
+
+  it('should allow forcing install with both NPM and Yarn', async () => {
+    await withFixture('blank-00', async (projectRoot) => {
+      const spawn = (cmd, args, opts) => {
+        expect(cmd).toBe('expo')
+        expect(args).toEqual([
+          'install',
+          '@bugsnag/plugin-expo-eas-sourcemaps',
+          '--npm',
+          '--yarn'
+        ])
+
+        expect(opts).toEqual({ cwd: projectRoot })
+
+        const proc = new EventEmitter()
+        // @ts-ignore
+        proc.stdout = new Readable({
+          read () {
+            this.push('some data on stdout')
+            this.push(null)
+          }
+        })
+        // @ts-ignore
+        proc.stderr = new Readable({
+          read () {
+            this.push('some data on stderr')
+            this.push(null)
+          }
+        })
```

Comment:
Do these tests or the underlying behavior depend on anything being emitted from stdout / stderr? What happens if there's nothing to read? (same q's for the other similar tests, aside from the ones asserting on the output contents)